src/main/java/uk/ac/ebi/ampt2d/commons/accession/persistence/BasicSpringDataRepositoryDatabaseService.java

```diff
@@ -89,4 +96,38 @@ public void enableAccessions(List<AccessionWrapper<MODEL, String, ACCESSION>> ac
                 .collect(Collectors.toSet()));
     }
 
+    @Override
+    public AccessionWrapper<MODEL, String, ACCESSION> update(AccessionWrapper<MODEL, String, ACCESSION> accession)
+            throws AccessionDoesNotExistException, HashAlreadyExistsException {
+        Collection<ENTITY> accessionedElements = repository.findByAccessionIn(Arrays.asList(accession.getAccession()));
+        assertAccessionExists(accession, accessionedElements);
+        assertHashDoesNotExist(accession);
```

Comment:
I reply to myself. First, this is not the variant accessioning project, but accessioning commons. Second, if the object is updated then its hash will change.

Reply:
No, if we call the accessioned method, it would be a new one, but if we call the update operation we need to check that it doesn't exist previously in the database.

Reply:
Yeah, we are in commons.
If the object is updated, then the content changes, thus hash changes, accession is reused.
You can't update an object with the content of an already existing accessioned object, this is this assert.